automated/linux/ota-update/static-delta.sh

```diff
@@ -0,0 +1,195 @@
+# auto register
+systemctl enable --now lmp-device-auto-register || true  # exit code will not be 0 as aklite is masked
+
+while ! journalctl --no-pager -u lmp-device-auto-register | grep "Device is now registered"
+do
+    echo "Waiting for device registration"
+    sleep 1
+done
+cat /var/sota/sota.toml
+df -B1
+# retrieve repo URL from sota.toml
+REPO_URL=$(grep repo_server /var/sota/sota.toml | awk '{gsub("\"","");print $3}')
+OSTREE_URL=$(grep ostree_server /var/sota/sota.toml | awk '{gsub("\"","");print $3}')
+. /etc/os-release
+curl -o targets.json -H "x-ats-tags:${LMP_FACTORY_TAG}" --cert /var/sota/client.pem --cacert /var/sota/root.crt --key /var/sota/pkey.pem "${REPO_URL}/targets.json"
+SEARCH_STRING=".signed.targets.\"${LMP_MACHINE}-lmp-${NEXT_TARGET}\".custom.\"delta-stats\".sha256"
+DELTA=$(jq -r "${SEARCH_STRING}" targets.json)
+if [ -z "${DELTA}" ]; then
+    error_msg "Static delta not available for target ${LMP_MACHINE}-lmp-${NEXT_TARGET}"
+    exit 0;
+fi
+curl -o delta.json -skL --cert /var/sota/client.pem --cacert /var/sota/root.crt --key /var/sota/pkey.pem "${OSTREE_URL}/delta-stats/${DELTA}"
+cat delta.json
+# obtain current OSTREE_SHA from ostree admin status
+ostree admin status
+CURRENT_OSTREE_SHA=$(ostree admin status | grep "\* lmp" | awk '{sub(/\..*/, ""); print $3}')
+if [ -z "${CURRENT_OSTREE_SHA}" ]; then
+    error_msg "ostree sha not found"
+    exit 0;
+fi
+SIZE_SEARCH_STRING=".[].\"${CURRENT_OSTREE_SHA}\".u_size"
+DELTA_SIZE=$(jq -r "${SIZE_SEARCH_STRING}" delta.json)
+if [ -z "${DELTA_SIZE}" ] || [ "${DELTA_SIZE}" = "null" ]; then
+    error_msg "Size of static delta for update from ${CURRENT_OSTREE_SHA} not found"
+    exit 0;
+fi
+
+# check how much disk is free
+DEVICE=$(df -h | grep "${ROOT}" | cut -d" " -f1)
+TOTAL=$(df -B1 --output=size,avail,target | grep "${ROOT}" | awk '{print $1}')
+FREE=$(df -B1 --output=size,avail,target | grep "${ROOT}" | awk '{print $2}')
+BLOCK_SIZE=$(dumpe2fs -h "${DEVICE}" | grep "Block size" | awk '{print $3}')
+
+# check what percentage of disk will the OTA take
+MAX_BYTES_AVAILABLE=$(echo "s=${TOTAL}*(${OTA_THRESHOLD}/100);scale=0;s/1" | bc -l)
+BYTES_IN_USE=$(echo "${TOTAL}-${FREE}" | bc -l)
+MAX_BYTES_AVAILABLE_FOR_UPDATE=$(echo "${MAX_BYTES_AVAILABLE}-${BYTES_IN_USE}" | bc -l)
+AFTER_OTA=$(echo "${MAX_BYTES_AVAILABLE_FOR_UPDATE}-${DELTA_SIZE}" | bc -l)
+
+if [ "${AFTER_OTA}" -lt 0 ]; then
+    error_msg "Disk already filled above threshold"
+    exit 0;
+fi
+
+# fill in disk so only ~1% is left after OTA (default value)
+# it is assumed that there is enough disk space to pefrorm the test
+# If FILL_SIZE > 100, the test will check if aklite aborts the OTA
+# Please note the calculations might be slightly wrong due to rounding
+# It is advised to use a larger number (>106) to force download failure
+TO_FILL=$(echo "s=${AFTER_OTA}*(${FILL_SIZE}/100)/${BLOCK_SIZE};scale=0;s/1" | bc -l)
+
+dd if=/dev/urandom of=fill_file bs="${BLOCK_SIZE}" count="${TO_FILL}"
+ls -l
+df -B1
```

Comment:
are these two lines needed?

Reply:
They're very helpful when reading the test job in LAVA. I can add a comment explaining why they're there but I'd like to keep them.